src/metrics-gatherer.ts

```diff
@@ -0,0 +1,230 @@
+		val : number,
+		labels : LabelSet = {}) {
+		this.histogram(`${name}_hist`, val, labels);
+		this.summary(`${name}_summary`, val, labels);		
+	}
+
+	// check that a metric is of the given type(s)
+	checkMetricType(name : string, kinds : string[]) {
+		return kinds.includes(this.kinds[name]);
+	}
+
+	// used declaratively to ensure a given metric of a certain kind exists, 
+	// given some custom params to instantiate it if absent
+	ensureExists(name : string, kind : string, custom : CustomParams = {}) {
+		// if exists, bail early
+		if (this.existMap[name]) {
+			return;
+		}
+		// if already exists with another kind, throw error
+		if (this.kinds[name] && this.kinds[name] != kind) {
+			throw new MetricsGathererError(`tried to use ${name} twice - first as ` +
+					`${this.kinds[name]}, then as ${kind}`);
+		}
+		// if not described, describe (poorly)
+		if (!(name in this.descriptions)) {
+			this.descriptions[name] = `undescribed ${kind} metric`;
+		}
+		// if doesn't exist, create metric
+		if (!this.metrics[kind][name]) {
+			const constructors : ConstructorMap = {
+				'gauge': new MetricConstructor(prometheus.Gauge),
+				'counter': new MetricConstructor(prometheus.Counter),
+				'summary': new MetricConstructor(prometheus.Summary),
+				'histogram': new MetricConstructor(prometheus.Histogram),
+			};
+			// mix provided custom params with custom params given by 
+			// `describe()`
+			custom = Object.assign(custom, this.customParams[name]);
+			this.metrics[kind][name] = constructors[kind].create({
+				name: name,
+				help: this.descriptions[name],
+				...custom
+			});
+			this.kinds[name] = kind;
+			this.existMap[name] = true;
+		}
+	}
+
+	// reset a given metric by name
+	reset(name: string) {
+		// will only have an entry in this.kinds if the metric has been observed 
+		// at least once (don't need to "reset" otherwise, and avoids error in
+		// indexing)
+		if (this.kinds[name]) {
+			this.metrics[this.kinds[name]][name].reset();
+		}
+	}
+
+	// create an express request handler given an auth test function
+	requestHandler(authTest? : AuthTestFunc) : express.Handler {
+		return (req : express.Request, res : express.Response) => {
+			if (authTest && !authTest(req)) {
+				return res.status(403).send();
+			} 
+			res.writeHead(200, { 'Content-Type': 'text/plain' });
+			res.end(prometheus.register.metrics());
+		};
+	}
+
+	aggregateRequestHandler(authTest? : AuthTestFunc) : express.Handler {
+		const aggregatorRegistry = new prometheus.AggregatorRegistry();
+		return (req, res) => {
+			if (authTest && !authTest(req)) {
+				return res.status(403).send();
+			}
+			aggregatorRegistry.clusterMetrics()
+				.then((metrics: string) => {
+					res.set('Content-Type', aggregatorRegistry.contentType);
+					res.send(metrics);
+				})
+				.catch((err: Error) => {
+					debug(`error in /cluster_metrics: ${err}`);
+					res.status(500).send();
+				});
+		};
+	}
+
+	// get the prometheus output
+	output() : string {
```

Comment:
Can you set up resin-lint/prettier for this repo please to get the auto-formatting